### PR TITLE
Added files for task 10

### PR DESCRIPTION
**Pull request by ArtemPurtoian**

### CODE REVIEW

File: task_10/building.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        return self.building_area
+
+    def number_of_floors(self):
+        return self.number_of_floors
+
+    def has_elevator(self):
+        return self.has_elevator
+
+    def building_material(self):
+        return self.building_material
```
Comment (bohdandovbsyh) [inline task_10/building.py:22]: why do you need all these 4 functions at all?

File: task_10/building_business.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class Business(Building):
+    def __init__(self, building_area, number_of_floors, has_elevator,
+                 building_material, has_parking_lot: bool):
+        super().__init__(building_area, number_of_floors, has_elevator,
+                         building_material)
+        self.has_parking_lot = has_parking_lot
+
+    def number_of_offices(self):
+        return self.number_of_offices
```
Comment (bohdandovbsyh) [inline task_10/building_business.py:12]: how does it work? you just return the function object. what for?

File: task_10/building_business.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    def has_parking_lot(self):
+        return self.has_parking_lot
```
Comment (bohdandovbsyh) [inline task_10/building_business.py:15]: same here...

File: task_10/building_government.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+class Government(Building):
+    def __init__(self, building_area, number_of_floors, has_elevator,
+                 building_material, department: str):
+        super().__init__(building_area, number_of_floors, has_elevator,
+                         building_material)
+        self.department = department
+
+    def department(self):
+        return self.department
```
Comment (bohdandovbsyh) [inline task_10/building_government.py:12]: what is it for?

File: task_10/business_bank.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class Bank(Business):
+    def __init__(self, building_area, number_of_floors, has_elevator,
+                 building_material, has_parking_lot, bank_name: str):
+        super().__init__(building_area, number_of_floors, has_elevator,
+                         building_material, has_parking_lot)
+        self.bank_name = bank_name
+
+    def bank_name(self):
```
Comment (bohdandovbsyh) [inline task_10/business_bank.py:11]: How does it work?

File: task_10/business_office.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+class Office(Business):
+    def __init__(self, building_area, number_of_floors, has_elevator,
+                 building_material, has_parking_lot, number_of_offices: int):
+        super().__init__(building_area, number_of_floors, has_elevator,
+                         building_material, has_parking_lot)
+        self.number_of_offices = number_of_offices
+
+    def number_of_offices(self):
+        return self.number_of_offices
```
Comment (bohdandovbsyh) [inline task_10/business_office.py:12]: What is it for?

File: task_10/government_courthouse.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+class Courthouse(Government):
+    def __init__(self, building_area, number_of_floors, has_elevator,
+                 building_material, department, number_of_courtrooms: int):
+        super().__init__(building_area, number_of_floors, has_elevator,
+                         building_material, department)
+        self.number_of_courtrooms = number_of_courtrooms
+
+    def courtrooms(self):
+        return self.courtrooms
```
Comment (bohdandovbsyh) [inline task_10/government_courthouse.py:12]: What is it for?

File: task_10/government_prison.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        super().__init__(building_area, number_of_floors, has_elevator,
+                         building_material, department)
+        self.number_of_cells = number_of_cells
+        self.max_prisoners = max_prisoners
+
+    def number_of_cells(self):
+        return self.number_of_cells
+
+    def max_prisoners(self):
+        return self.max_prisoners
```
Comment (bohdandovbsyh) [inline task_10/government_prison.py:17]: What is it for?

File: task_10/government_prison.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    def __init__(self, building_area, number_of_floors, has_elevator,
+                 building_material, department, number_of_cells: int,
+                 max_prisoners: int):
+        super().__init__(building_area, number_of_floors, has_elevator,
+                         building_material, department)
+        self.number_of_cells = number_of_cells
+        self.max_prisoners = max_prisoners
+
+    def number_of_cells(self):
+        return self.number_of_cells
```
Comment (bohdandovbsyh) [inline task_10/government_prison.py:14]: What is it for?

File: task_10/residential_apartment.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+class ApartmentBuilding(Residential):
+    def __init__(self, building_area, number_of_floors, has_elevator,
+                 building_material, has_garden, number_of_apartments: int):
+        super().__init__(building_area, number_of_floors, has_elevator,
+                         building_material, has_garden)
+        self.number_of_apartments = number_of_apartments
+
+    def number_of_apartments(self):
+        return self.number_of_apartments
```
Comment (bohdandovbsyh) [inline task_10/residential_apartment.py:12]: What is it for?

File: task_10/residential_house.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        self.has_garage = has_garage
+
+    def number_of_bedrooms(self):
+        return self.number_of_bedrooms
+
+    def has_swimming_pool(self):
+        return self.has_swimming_pool
+
+    def has_garage(self):
+        return self.has_garage
```
Comment (bohdandovbsyh) [inline task_10/residential_house.py:21]: What is it for?  all 3

File: task_10/building.py (hunk 1 above)
Comment (ArtemPurtoian) [reply task_10/building.py:22]: I didn't even check the output without these functions. Just thought they need to be there along with an abstractmethod.
Seems that I was wrong and they are completely redundant as I define all attributes in __init__ and they are inherited.
Got it.
All works fine without unnecessary functions.
Screenshot 2023-06-15 130907

Comment (bohdandovbsyh) [reply task_10/building.py]: Please check it in future))

**Review by bohdandovbsyh — APPROVED** [review]

**Pull request merged**